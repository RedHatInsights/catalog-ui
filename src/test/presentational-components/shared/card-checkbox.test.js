import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CardCheckbox from '../../../presentational-components/shared/card-checkbox';

describe('<CardCheckbox />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      handleCheck: jest.fn(),
      isChecked: false,
      id: 'foo-id'
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<CardCheckbox { ...initialProps }/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleCheck on checkbock click', () => {
    const wrapper = mount(<CardCheckbox { ...initialProps }/>);
    wrapper.find('svg').simulate('click');
    expect(initialProps.handleCheck).toHaveBeenCalledWith(expect.any(Object));
  });
});
