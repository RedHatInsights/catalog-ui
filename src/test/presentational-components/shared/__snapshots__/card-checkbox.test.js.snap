// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCheckbox /> should render correctly 1`] = `
<div
  style={
    Object {
      "float": "right",
    }
  }
>
  <OutlinedSquareIcon
    aria-label="card checkbox"
    className="icon-checkbox "
    color="currentColor"
    id="foo-id"
    onClick={[MockFunction]}
    size="sm"
    title={null}
  />
</div>
`;
